descwl_coadd/coadd.py

```diff
@@ -276,11 +412,150 @@ def make_coadd(
+    warper=None, mfrac_warper=None,
+):
+    """
+    make warps from the input exposure, including warps for the PSF, noise
+    images and masked fraction
+
+    This is the entry point to the package from the LSST Pipetask.
+
+    Parameters
+    ----------
+    exp: ExposureF or DeferredDatasetHandle
+        Either an Exposure or a corresponding DeferredDatasetHandle
+    coadd_wcs: DM wcs object
+        The target wcs
+    coadd_bbox: geom.Box2I
+        The bounding box for the coadd within larger wcs system
+    psf_dims: tuple
+        The dimensions of the psf
+    rng: np.random.RandomState
+        The random number generator for making noise images
+    remove_poisson: bool
+        If True, remove the poisson noise from the variance
+        estimate.
+    warper: afw_math.Warper
+        The warper to use for the image, noise, and psf
+    mfrac_warper: afw_math.Warper
+        The warper to use for the masked fraction
+
+    Returns
+    -------
+    warp, noise_warp, psf_warp, mfrac_warp, exp_info
+
+    TODO
+    ----
+        - make a list of noise exposures
+    """
+
+    # this is the requested coadd psf dims
+    check_psf_dims(psf_dims)
+
+    # Get integer center of coadd and corresponding sky center.  This is used
+    # to construct the coadd psf bounding box and to reconstruct the psfs
+    coadd_cen_integer, coadd_cen_skypos = get_coadd_center(
+        coadd_wcs=coadd_wcs, coadd_bbox=coadd_bbox,
+    )
+
+    coadd_psf_bbox = get_coadd_psf_bbox(cen=coadd_cen_integer, dim=psf_dims[0])
+    coadd_psf_wcs = coadd_wcs
+
+    # can re-use the warper for each coadd type except the mfrac where we use
+    # linear
+    if warper is None:
+        warp_config = afw_math.Warper.ConfigClass()
+        warp_config.warpingKernelName = DEFAULT_INTERP
+        warper = afw_math.Warper.fromConfig(warp_config)
+
+    if mfrac_warper is None:
+        warp_config = afw_math.Warper.ConfigClass()
+        warp_config.warpingKernelName = "bilinear"
+        mfrac_warper = afw_math.Warper.fromConfig(warp_config)
+
+    # will zip these with the exposures to warp and add
+    wcss = [coadd_wcs, coadd_wcs, coadd_psf_wcs, coadd_wcs]
+    bboxes = [coadd_bbox, coadd_bbox, coadd_psf_bbox, coadd_bbox]
+    warpers = [warper, warper, warper, mfrac_warper]
+
+    # PSF will generally have NO_DATA in warp as we tend to use the same psf
+    # stamp size for input and output psf and just zero out wherever there is
+    # no data
+
+    verifys = [False]*4  # HACK [True, True, False, True]
+
+    LOG.info('warping and adding exposures')
+
+    if isinstance(exp, DeferredDatasetHandle):
+        expobj = exp.get()
+    else:
+        expobj = exp
+
+    # TODO use exp.getId() when it arrives in weekly 44.  For now use an index
+    # 0::len(exps)
+    try:
+        exp_id = expobj.getId()
+    except AttributeError:
+        exp_id = -9999
+
+    bad_msk, maskfrac = get_bad_mask(expobj)
```

Comment:
I don't know how useful this `maskfrac` is, since it computes it for the entire `calexp`, regardless whether it is within the patch's boundary or not. May be if we restricted this within the bounding box of the `warp` or a `coadd`, that'd be more useful.

Reply:
yes, this should be calculated on the warp trimmed to the cell coadd region.

I do think we want to remove highly masked (trimmed) images